etc/makefile.main

```diff
@@ -19,11 +19,12 @@ FASTTRAX_DIR := FastTrax
 OPTIMIZER		=	-O3
 
 CC		     	:=	gcc
-CFLAGS			 =	$(OPTIMIZER) -Wall -Wno-multichar -Wno-ctor-dtor-privacy
+CFLAGS			 =	$(OPTIMIZER) -Wall -Wno-multichar -Wno-ctor-dtor-privacy -D__USE_GNU -DSTDC_HEADERS -D_POSIX_SOURCE
```

Comment:
```suggestion
CFLAGS			 =	$(OPTIMIZER) -Wall -Wno-multichar -Wno-ctor-dtor-privacy -D_DEFAULT_SOURCE -DSTDC_HEADERS -D_POSIX_SOURCE
```

Usually setting the right -std option is better. see https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html
For instance -std=gnu11

Reply:
Maybe include features.h then it should define __USE_GNU as needed.

Reply:
Right, _GNU_SOURCE should imply __USE_GNU, but that missing in features.h

Reply:
I would take a different approach. `__USE_GNU` is glibc specific and I didn't expect it would be used in public headers. So headers/posix/regex.h should be adapted to use our variant of features.h, which use internal defines that are not compatible with glibc.

It seems this header is the only one in headers/posix that was imported directly from glibc? (or maybe there are more, but nothing that uses `__USE_GNU`). Usually our public headers are written by ourselves, and glibc headers, if we use any, are kept private (in sources/system/libroot) and not available for applications.

Reply:
yes, whichever fix we do in Haiku in the end, it will probably be only for the next release anyway. Until then, I think defining `__USE_GNU` here is fine.